README.md

```diff
@@ -17,3 +17,100 @@ Starting Point
 The [Act](https://github.com/dapphub/klab/blob/master/acts.md) specification language already matches some of the desired features: It has a simple structure, simple logic, and supports properties written for EVM bytecode.
 
 It is currently being used as the starting point for this project, requiring few modifications to the structure and syntax such that it can also support high level properties.
+
+Infrastructure
+==============
+The grammar for the specification language is in the `src` repository. This is the front end parsing of the language. Given set of `act` behaviours (transitions), one can generate a set of proof obligations, expressed as a json object:
+```json
+[{"name": "Transfer_case0",
+ "contract": "Token",
+ "status_code": "EVMC_SUCCESS",
+ "return": [],
+ "case": "pass",
+ "bytecode": "0xdeadbeef",
+ "storageLayout": "probably",
+ "interface": "the same as in the spec",
+ "storage_pre": [
+     {"type": "uint256",
+      "name": "balanceOf[msg.sender]",
+      "value": "v0"
+     },
+     {"type": "uint256",
+      "name": "balanceOf[to]",
+      "value": "v1"
+     }
+ ],
+ "storage_post": [
+     {"type": "uint256",
+      "name": "balanceOf[msg.sender]",
+      "value": "v0 - amount"
+     },
+     {"type": "uint256",
+      "name": "balanceOf[to]",
+      "value": "v1 + amount"
+     }
+ ],
+ "precondition": [
+   "CALLER =/= to",                                 // case specific
+   "CALLVALUE == 0",                                // from iff
+   "value <= balanceOf[CALLER]",
+   "CALLER =/= to => balanceOf[to] + value < 2^256",//
+   "0 <= to",                                       //from type restrictions
+   "to < 2^160",
+   "0 <= v0",
+   "v0 < 2^256",
+   "0 <= v1",
+   "v0 < 2^256",
+   "v0 - amount <= v0",
+   "v0 - amount < 2^256",
+   "v1 + amount <= v0",
+   "v1 + amount < 2^256"
+ ],
+ "postcondition": []
```

Comment:
So in this format where each JSON object is a case, the postcondition would be repeated in each object, right?

Reply:
Yes